### PR TITLE
[Frontend] CSVMapper - fix missing value in relationship related-to a…

### LINKED ISSUE #4797: CSV Mapper prevent related-to relationships + missing revoked by
## Description

This is 2 problems on generic relationships.

**related to**
When trying to create a csv mapper with related-to relationships, the available options have weird behavior and prevent the configuration of this kind of relation.

**revoked-by**
When trying to create a csv mapper with related-to relationships, nothing is available for selection

## Environment

1. OpenCTI version: 5.11.12
2. OpenCTI client: frontend

## Reproducible Steps

Steps to create the smallest reproducible scenario:
1. Create a fresh CSV Mapper
2. Add first entity Area
3. Add second entity Cryptocurrency-wallet
4. Create a relationship representation "related-to"
5. Try to link both entity representation

## Expected Output

related-to is compatible with everything so we should be able to configure Area-related to-Crypto or Crypto-related to-Area
This must be also possible for any types available.

## Actual Output

Strange behavior. Option can be available in the second box but not in the first one. When one part is selected, there is no more option in the first one.

image

## Code hints

For missing handling of revoked-by, Relation.ts

`if (values.includes(relationshipType) || relationshipType === 'related-to') {`

### PR DESCRIPTION
**Pull request by jpkha**
### Proposed changes

* Fixed relationship related-to and revoked-by

### Related issues

* https://github.com/OpenCTI-Platform/opencti/issues/4797

### Checklist

- [x] I consider the submitted work as finished
- [x] I tested the code for its functionality
- [x] I wrote test cases for the relevant uses case
- [x] I added/update the relevant documentation (either on github or on notion)
- [x] Where necessary I refactored code to improve the overall quality

### CODE REVIEW

File: opencti-platform/opencti-front/src/private/components/data/csvMapper/representations/attributes/CsvMapperRepresentationAttributeRefForm.tsx [hunk 1]
```diff
@@ -107,7 +93,31 @@ CsvMapperRepresentationAttributeRefFormProps
         return r.id !== fromId;
       }
       return true;
-    });
+    }));
+
+  let options;
+  if (entityType === 'related-to' || entityType === 'revoked-by') {
+    options = filteredOptions(representations);
+  } else {
```
Comment (lndrtrbn) [inline opencti-platform/opencti-front/src/private/components/data/csvMapper/representations/attributes/CsvMapperRepresentationAttributeRefForm.tsx:104]: Maybe add a comment to explain why we are doing this? 
From an external point of view (me) this is black magic 🧙

Comment (jpkha) [reply opencti-platform/opencti-front/src/private/components/data/csvMapper/representations/attributes/CsvMapperRepresentationAttributeRefForm.tsx:104]: Yes, good point @lndrtrbn

Comment (jpkha) [reply opencti-platform/opencti-front/src/private/components/data/csvMapper/representations/attributes/CsvMapperRepresentationAttributeRefForm.tsx:104]: I add this comment : 
```
// For both entity types, whether they are related-to or revoked-by,
// we don't need to resolve the different types, as they can link any entity between them.
```

Is it clearer with this comment ?

File: opencti-platform/opencti-front/src/private/components/data/csvMapper/representations/attributes/CsvMapperRepresentationAttributeRefForm.tsx [hunk 2]
```diff
@@ -107,7 +93,34 @@ CsvMapperRepresentationAttributeRefFormProps
         return r.id !== fromId;
       }
       return true;
-    });
+    }));
+
+  let options;
+  
+// For both entity types, whether they are related-to or revoked-by,
+// we don't need to resolve the different types, as they can link any entity between them.
```
Comment (lndrtrbn) [inline opencti-platform/opencti-front/src/private/components/data/csvMapper/representations/attributes/CsvMapperRepresentationAttributeRefForm.tsx]:
```suggestion
  // For both entity types, whether they are related-to or revoked-by,
  // we don't need to resolve the different types, as they can link any entity between them.
```
To fix eslint errors

**Review by lndrtrbn — APPROVED** [review]

**Pull request merged**